File: Sorting/01-Bubble-Sort.py
def bubbleSort(arr):
        n = len(arr)
        for i in range(n-1):
            swapped = False
            for j in range(n-1-i):
                if(arr[j] > arr[j + 1]):
                    # swap
                    arr[j], arr[j + 1] = arr[j + 1], arr[j]
                    swapped = True
            if not swapped: break # Already Sorted
                    
def main():
    arr = [2, 22, 54, 87, 1, 11, 54, 90, 10]
    print("Array Before Sorting: ", arr)
    bubbleSort(arr)
    print("Array After Sorting: ", arr)

main()

'''
Array Before Sorting:  [2, 22, 54, 87, 1, 11, 54, 90, 10]
Array After Sorting:  [1, 2, 10, 11, 22, 54, 54, 87, 90]
'''

File: Sorting/01-Bubble-Sort.js
const bubbleSort = (arr) => {
  let n = arr.length;
  for (let i = 0; i < n; i++) {
    let swapped = false;
    for (let j = 0; j < n - 1 - i; j++) {
      if (arr[j] > arr[j + 1]) {
        // Swap
        let temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        swapped = true;
      }
    }
    // Already Sorted
    if (!swapped) {
      break;
    }
  }
};

function main() {
  let arr = [22, 1, 43, 53, 52, 7, 63, 9, 10];
  console.log(`Array Before Sorting: ${arr}`);
  bubbleSort(arr);
  console.log(`Array Before Sorting: ${arr}`);
}

main();

/** output
    Array Before Sorting: 22,1,43,53,52,7,63,9,10
    Array Before Sorting: 1,7,9,10,22,43,52,53,63
 */
